prov/gni/include/gnix_nic.h

```diff
@@ -296,16 +326,18 @@ struct gnix_smsg_amo_cntr_hdr {
  */
 struct gnix_tx_descriptor {
 	struct dlist_entry          list;
+	struct slist_entry	    slist;
```

Comment:
Please add the description of what this is used for (and why you need another list element) above in doxygen comments.

Also I have not traced the use of the `list` and `slist` fields throughout, but is it possible to just use `list` or are they in use simultaneously?